maven/src/main/java/org/owasp/dependencycheck/maven/BaseDependencyCheckMojo.java

```diff
@@ -2212,46 +2212,17 @@ protected void populateSettings() {
                 if (mavenProxy.getUsername() != null && !mavenProxy.getUsername().isEmpty()) {
                     System.setProperty("https.proxyUser", mavenProxy.getUsername());
                 }
-                if (mavenProxy.getPassword() != null && !mavenProxy.getPassword().isEmpty()) {
-                    System.setProperty("https.proxyPassword", mavenProxy.getPassword());
-                }
-                if (mavenProxy.getNonProxyHosts() != null && !mavenProxy.getNonProxyHosts().isEmpty()) {
-                    System.setProperty("http.nonProxyHosts", mavenProxy.getNonProxyHosts());
-                }
-            }
+                String password = mavenProxy.getPassword();
+                if (password != null && !password.isEmpty()) {
+                    try {
+                        password = decryptPasswordFromSettings(password);
+                    } catch (SecDispatcherException ex) {
+                        password = handleSecDispatcherException("proxy", mavenProxy.getId(), password, ex);
+                    }
 
-            settings.setString(Settings.KEYS.PROXY_SERVER, mavenProxy.getHost());
-            settings.setString(Settings.KEYS.PROXY_PORT, Integer.toString(mavenProxy.getPort()));
-            final String userName = mavenProxy.getUsername();
-            String password = mavenProxy.getPassword();
-            if (password != null && !password.isEmpty()) {
-                if (settings.getBoolean(Settings.KEYS.PROXY_DISABLE_SCHEMAS, true)) {
-                    System.setProperty("jdk.http.auth.tunneling.disabledSchemes", "");
-                }
-                try {
-                    password = decryptPasswordFromSettings(password);
-                } catch (SecDispatcherException ex) {
-                    password = handleSecDispatcherException("proxy", mavenProxy.getId(), password, ex);
+                    System.setProperty("https.proxyPassword", password);
                 }
             }
```

Comment:
Your change breaks configuring the nonProxyHosts in the maven settings.xml
I think it should be restored (would've added it as a suggestion for the line where setting it in the legacy settings is deleted, but suggestions on deleted code are not supported by Github)

```suggestion
                final String nonProxyHosts = mavenProxy.getNonProxyHosts();
                if (nonProxyHosts != null && !nonProxyHosts.isEmpty()) {
                    System.setProperty("http.nonProxyHosts", nonProxyHosts);
                }
            }
```

Reply:
Oh dear, given this is actually the use case I wanted to fix first place :see_no_evil: The line got lost with the last minute password change. Good to have a second pair of eyes on a new day.